packages/headless/src/features/analytics/analytics-utils.ts

```diff
@@ -32,19 +32,19 @@ export type SearchAction = AsyncThunkAction<
   {analyticsType: AnalyticsType.Search},
   void | {},
   AsyncThunkAnalyticsOptions<StateNeededByAnalyticsProvider>
->;
+> & {name: string};
```

Comment:
One of the major issues I had was that `SearchAction`, when building the project, was being replaced with its resulting type.

The only way I found of preventing the typescript transpiler from skipping `SearchAction` in favor of its definition was to add an intersection with an existing property (`name` legitimately is part of this type already).

Reply:
Hmm, I am not keen on doing this for all our async actions. Could you give an example of the action you are trying to resolve, and the resolved entity you want to construct from it.

Reply:
I was trying to resolve the type of `executeSearch` as
```ts
const executeSearch: AsyncThunkDefinition<
  'search/executeSearch',
  ExecuteSearchThunkReturn,
  SearchAction,
  AsyncThunkSearchOptions<StateNeededByExecuteSearch>;
```
however, without the addition of `& {name: string}`, it was resolved as
```ts
const executeSearch: AsyncThunkDefinition<
  'search/executeSearch',
  ExecuteSearchThunkReturn,
  AsyncThunkAction<
    {analyticsType: AnalyticsType.Search},
    void | {},
    AsyncThunkAnalyticsOptions<StateNeededByAnalyticsProvider>
  >,
  AsyncThunkSearchOptions<StateNeededByExecuteSearch>;
```